src/compiler/checker.ts

```diff
@@ -3993,6 +4027,37 @@ namespace ts {
             return links.resolvedType;
         }
 
+        // We do not perform supertype reduction on intersection types. Intersection types are created only by the &
+        // type operator and we can't reduce those because we want to support recursive intersection types. For example,
+        // a type alias of the form "type List<T> = T & { next: List<T> }" cannot be reduced during its declaration.

```

Comment:
Yes, the primary reason is the observable differences in call/construct signatures. For example, #2610. With intersection types you always get the combined set of signatures so there is no loss of information and supertype reduction wouldn't make an observable difference.